notanorm/base.py

```diff
@@ -403,14 +405,31 @@ def capture_sql(
             self.__capture = False
 
     def create_model(self, model: DbModel, ignore_existing=False):
+        """Given a database model, create all the corresponding tables and indexes."""
         for name, schema in model.items():
             self.create_table(name, schema, ignore_existing)
 
-    def create_table(self, name, schema: DbTable, ignore_existing=False):
-        raise RuntimeError("Generic models not supported")
+    def create_table(
+        self, name, schema: DbTable, ignore_existing=False
+    ):  # pragma: no cover
+        raise RuntimeError("Generic create table not supported")
 
-    def create_indexes(self, name, schema: DbTable, ignore_existing=False):
-        raise RuntimeError("Generic models not supported")
+    def create_indexes(self, name, schema: DbTable):  # pragma: no cover
+        raise RuntimeError("Generic create index not supported")
+
+    def drop_index(self, table: str, index: DbIndex):
+        for idx in self.model()[table].indexes:
+            if idx == index:
+                self.drop_index_by_name(table, index.name)
+
+    def drop_index_by_name(self, table: str, index_name):
+        sql = "drop index " + self.quote_key(index_name)
+        self.execute(sql)
+
+    @staticmethod
+    def unique_index_name(table, field_names):
+        # prefer urandom to secrets, less opaque
+        return "ix_" + table + "_" + "_".join(field_names) + "_" + os.urandom(16).hex()
```

Comment:
Is there a length limit to an index name that might be overrun by joining the field names together?

Reply:
I think it's 62 characters

Reply:
mysql 8.0 : no meaningful limit (1gb i think).  sqlite :  1000000 chars, postgres: 63 chars.  drivers can override "unique_index_name" (which has been deliberately moved to db/base.py if needed

Reply:
imo: mysql 5.7 and postgres drivers should use shorter names (ix_table_uuid should be fine)